vmm/src/config.rs

```diff
@@ -2157,12 +2187,77 @@ impl RestoreConfig {
             .map_err(Error::ParseRestore)?
             .unwrap_or(Toggle(false))
             .0;
+        let net_ids = parser
+            .convert::<StringList>("net_ids")
+            .map_err(Error::ParseRestore)?
+            .map(|v| v.0);
+        let net_fds = parser
+            .convert::<IntegerList>("net_fds")
+            .map_err(Error::ParseRestore)?
+            .map(|v| v.0.iter().map(|e| *e as i32).collect());
 
         Ok(RestoreConfig {
             source_url,
             prefault,
+            net_ids,
+            net_fds,
         })
     }
+
+    pub fn validate(&self, vm_config: &VmConfig) -> ValidationResult<()> {
+        // Check if multiple same ids are passed
+        // verify if passed ids are valid
+        let vm_net_ids = match &vm_config.net {
+            // It's okay to unwrap id because it is guaranteed to have a net id in the snapshot config.
+            Some(net_configs) => net_configs
+                .iter()
+                .map(|v| v.id.as_ref().unwrap().clone())
+                .collect(),
+            None => Vec::new(),
+        };
+        if let Some(net_ids) = &self.net_ids {
+            let mut net_ids_set = BTreeSet::new();
+            for net_id in net_ids.iter() {
+                if !vm_net_ids.contains(net_id) {
+                    return Err(ValidationError::InvalidIdentifier(net_id.clone()));
+                }
+                if !net_ids_set.insert(net_id.clone()) {
+                    return Err(ValidationError::IdentifierNotUnique(net_id.clone()));
+                }
+            }
+        }
+        // Iterate through the net_ids and check if all required ids are passed without any unexpected ids
+        // Also, verify that the number of fds passed is equal to the number of fds required
```

Comment:
With the syntax mentioned above, you should explicitly check if the number of FDs from restore request matches with the number of FDs needed of the corresponding net device (via `net_config.fds`).

Reply:
Its done with the latest approach.